src/Compilers/Core/Portable/Operations/ControlFlowConditionKind.cs

```diff
@@ -3,12 +3,30 @@
 namespace Microsoft.CodeAnalysis.FlowAnalysis
 {
     /// <summary>
-    /// PROTOTYPE(dataflow): Add documentation
+    /// Represents kind of conditional branch from a <see cref="BasicBlock"/>.
     /// </summary>
     public enum ControlFlowConditionKind
     {
+        /// <summary>
+        /// Indicates no conditional branch from a <see cref="BasicBlock"/>.
+        /// Associated <see cref="BasicBlock.ConditionalSuccessor"/> is null.
+        /// </summary>
         None,
+
+        /// <summary>
+        /// Indicates a conditional branch from a <see cref="BasicBlock"/>,
+        /// with a non-null <see cref="BasicBlock.BranchValue"/> and <see cref="BasicBlock.ConditionalSuccessor"/>.
+        /// If <see cref="BasicBlock.BranchValue"/> evaluates to <code>false</code>,
+        /// then the branch <see cref="BasicBlock.ConditionalSuccessor"/> is taken.
+        /// </summary>
         WhenFalse,
```

Comment:
Not related to this PR: Why do we have two kinds (when true and when false), given that we could represent all branches with just one (when true) by swapping the conditional branch and the fallthrough?

Reply:
We decided in the design meeting that `FallThroughSuccessor` is always a non-conditional branch. The Condition and the ConditionKind only apply to the `ConditionalSuccessor`, hence the need for two kinds.

Reply:
As we discussed offline, I'm not proposing to change the meaning of the fallthrough case. In `if (condition) { /*go to conditional branch*/ } /* fallthrough branch */`, the fallthrough branch only comes into play if the condition didn't match.

Reply:
@jcouv Stopped by and explained he was not proposing changing these semantics. His proposal was to change our CFG builder implementation so that we always imply `ConditionKind.WhenTrue` for the conditional branch, which means we can replace this enum with just a bool `IsConditional` in the basic block.

@AlekseyTs @tmat @DustinCampbell @gafter @333fred your thoughts on this proposal?

Reply:
We discussed this briefly at the last meeting and decided that the existing ability to put the two branches in either order permits us to more closely model the order of operations and the kind of code that would be generated, even though the semantics based on inverting labels would be the same. For example, if we're converting IL into a control-flow graph, we would not want to change the sense of the conditional and swap the labels, but rather reflect more directly the code as expressed in IL.